src/test/java/com/google/sps/AlgorithmTest.java

```diff
@@ -0,0 +1,293 @@
+    placeToVisitE.setProperty("place-name", "Place To Visit E");
+    placeToVisitE.setProperty("latitude", 40.6);
+    placeToVisitE.setProperty("longitude", -0.6);
+    placeToVisitE.setProperty("category", "places-to-visit");
+    placeToVisitE.setProperty("description", "A charming place to visit.");
+    placeToVisitE.setProperty("cost-rating", 1);
+    placeToVisitE.setProperty("crowd-rating", 2);
+    Entity placeToVisitF = new Entity("Recommendation");
+    placeToVisitF.setProperty("place-name", "Place To Visit F");
+    placeToVisitF.setProperty("latitude", 40.7);
+    placeToVisitF.setProperty("longitude", -0.7);
+    placeToVisitF.setProperty("category", "places-to-visit");
+    placeToVisitF.setProperty("description", "A wonderful place to visit.");
+    placeToVisitF.setProperty("cost-rating", 2);
+    placeToVisitF.setProperty("crowd-rating", 1);
+    // Store recommendations in datastore
+    ds.put(restaurantA);
+    ds.put(restaurantB);
+    ds.put(restaurantC);
+    ds.put(restaurantD);
+    ds.put(restaurantE);
+    ds.put(studyPlaceA);
+    ds.put(studyPlaceB);
+    ds.put(placeToVisitA);
+    ds.put(placeToVisitB);
+    ds.put(placeToVisitC);
+    ds.put(placeToVisitD);
+    ds.put(placeToVisitE);
+    ds.put(placeToVisitF);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testAllFiveReturned() throws Exception{
+    // When there are exactly five recommendations of that category, the algorithm should return  
+    // all 5 recommendations. The coordinate formed by (cost-rating, crowd-rating) should be closest
+    // to the coordinate of user preference for the first recommendation, then respectively for 
+    // the remaining 4 (lowest Euclidean distance)
+
+    // Mock request and mock response
+    HttpServletRequest request = mock(HttpServletRequest.class);
+    HttpServletResponse response = mock(HttpServletResponse.class);
+    when(request.getParameter("category")).thenReturn("restaurants");
+    when(request.getParameter("cost-rating")).thenReturn("3");
+    when(request.getParameter("crowd-rating")).thenReturn("3");
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+    myServlet.doGet(request, response);
+    writer.flush();
+    // 'Top 5' recommendations, ordered using the Euclidean distance (ED) algorithm
+    Collection<Recommendation> expected = Arrays.asList(
+      new Recommendation("Restaurant C", "restaurants", 53.0, -0.3, "A wonderful place to eat", 3, 3), // ED = 0
+      new Recommendation("Restaurant B", "restaurants", 52.0, -0.2, "A lovely place to eat", 2, 2), // ED = 1
+      new Recommendation("Restaurant D", "restaurants", 54.0, -0.4, "A charming place to eat", 4, 4), // ED = 1
+      new Recommendation("Restaurant A", "restaurants", 51.0, -0.1, "A great place to eat", 1, 1), // ED = sqrt(8) = 2.828...
```

Comment:
I believe that we don't compare square roots in the algorithm [1] given that `if sqrt(a) < sqrt(b) then a < b and vise versa`. Please update the comments with ED = sqrt(...).

[1] https://github.com/googleinterns/step241-2020/blob/master/src/main/java/com/google/sps/servlets/RecommenderServlet.java#L122

Reply:
Good point - thanks!